incubator/jaeger/templates/collector-deploy.yaml

```diff
@@ -31,41 +32,76 @@ spec:
       nodeSelector:
 {{ toYaml .Values.collector.nodeSelector | indent 8 }}
       containers:
-      - name: "{{ template "jaeger.fullname" . }}-collector"
-        image: "{{ .Values.collector.image }}:{{ .Values.collector.tag }}"
+      - name: {{ template "jaeger.fullname" . }}-collector
+        image: {{ .Values.collector.image }}:{{ .Values.collector.tag }}
         imagePullPolicy: {{ .Values.collector.pullPolicy }}
+        env:
+          - name: SPAN_STORAGE_TYPE
+            valueFrom:
+              configMapKeyRef:
+                name: {{ template "jaeger.fullname" . }}
+                key: span-storage.type
+          {{- if eq .Values.storage.type "cassandra" }}
+          - name: CASSANDRA_SERVERS
+            valueFrom:
+              configMapKeyRef:
+                name: {{ template "jaeger.fullname" . }}
+                key: cassandra.servers
+          - name: CASSANDRA_PORT
+            valueFrom:
+              configMapKeyRef:
+                name: {{ template "jaeger.fullname" . }}
+                key: cassandra.port
+          - name: CASSANDRA_KEYSPACE
+            valueFrom:
+              configMapKeyRef:
+                name: {{ template "jaeger.fullname" . }}
+                key: cassandra.keyspace
+          {{- end }}
+          {{- if eq .Values.storage.type "elasticsearch" }}
+          - name: ES_PASSWORD
+            valueFrom:
+              configMapKeyRef:
+                name: {{ template "jaeger.fullname" . }}
+                key: es.password
+          - name: ES_SERVER_URLS
+            valueFrom:
+              configMapKeyRef:
+                name: {{ template "jaeger.fullname" . }}
+                key: es.server-urls
+          - name: ES_USERNAME
+            valueFrom:
+              configMapKeyRef:
+                name: {{ template "jaeger.fullname" . }}
+                key: es.username
+          {{- end }}
+          - name: COLLECTOR_PORT
+            valueFrom:
+              configMapKeyRef:
+                name: {{ template "jaeger.fullname" . }}
+                key: collector.port
+          - name: COLLECTOR_HTTP_PORT
+            valueFrom:
+              configMapKeyRef:
+                name: {{ template "jaeger.fullname" . }}
+                key: collector.http-port
+          - name: COLLECTOR_ZIPKIN_HTTP_PORT
+            valueFrom:
+              configMapKeyRef:
+                name: {{ template "jaeger.fullname" . }}
+                key: collector.zipkin.http-port
         ports:
-        - containerPort: 14267
+        - containerPort: {{ .Values.collector.service.tchannelPort }}
           name: tchannel
           protocol: TCP
-        - containerPort: 14268
+        - containerPort: {{ .Values.collector.service.httpPort }}
           name: http
           protocol: TCP
-        - containerPort: 9411
+        - containerPort: {{ .Values.collector.service.zipkinPort }}
```

Comment:
@ledor473 
I used the named port from your PR, but the values are no longer hard-coded. Now they come from the `values.yaml` and are mapped in the config map.

Reply:
@pavelnikolov yes that is even better considering the Collector Pod will listen on that port instead of doing port-forwarding at the Service level 👍